### PR TITLE
Don't pass IAM role to ECS Service when there are more than one Target Groups

### PR DESCRIPTION
**Pull request by msmagoo87**
## what
* In the case of registering more than one Target Group, don't create or pass the ECS service IAM role to the `aws_ecs_service` block.

## why
* If you are registering multiple target groups to the service, you must allow ECS to create a service-linked role rather than passing one that was manually created. Passing one results in an API error from ECS: `Error: InvalidParameterException: You cannot specify an IAM role for services that require a service linked role.`

## references
* https://docs.aws.amazon.com/AmazonECS/latest/developerguide/register-multiple-targetgroups.html

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,5 +1,6 @@
 locals {
   enabled = module.this.enabled
+  enable_ecs_service_role = local.enabled && var.network_mode != "awsvpc" && length(var.ecs_load_balancers) <= 1
```
Comment (jamengual) [inline main.tf:3]: please rely in `module.this.enabled` instead of the local

File: main.tf [hunk 2]
```diff
@@ -274,7 +275,7 @@ resource "aws_ecs_service" "ignore_changes_task_definition" {
   platform_version                   = var.launch_type == "FARGATE" ? var.platform_version : null
   scheduling_strategy                = var.launch_type == "FARGATE" ? "REPLICA" : var.scheduling_strategy
   enable_ecs_managed_tags            = var.enable_ecs_managed_tags
-  iam_role                           = var.network_mode != "awsvpc" ? join("", aws_iam_role.ecs_service.*.arn) : null
+  iam_role                           = var.network_mode != "awsvpc" && length(var.ecs_load_balancers) <= 1 ? join("", aws_iam_role.ecs_service.*.arn) : null
```
Comment (aknysh) [inline main.tf:278]:
```suggestion
  iam_role                           = local.enable_ecs_service_role ? join("", aws_iam_role.ecs_service.*.arn) : null
```

File: main.tf [hunk 3]
```diff
@@ -356,7 +357,7 @@ resource "aws_ecs_service" "default" {
...
```
Comment (aknysh) [inline main.tf:360]:
```suggestion
  iam_role                           = local.enable_ecs_service_role ? join("", aws_iam_role.ecs_service.*.arn) : null
```

**Review by aknysh — COMMENTED** [review]
please see comments

**Review by jamengual — APPROVED** [review]

**Pull request merged**